Lib/test/test_contextlib.py

```diff
@@ -1117,9 +1117,15 @@ def test_cm_is_reentrant(self):
 
 
 class TestChdir(unittest.TestCase):
+    def make_relative_path(self, *parts):
+        return os.path.join(
+            os.path.dirname(os.path.normpath(__file__)),
```

Comment:
`os.path.normpath()` has a design flaw: it does not work in case of links. For example:
```
ln -s Lib lib2
./python lib2/test/test_contextlib.py
```
You need to use `realpath()`.

Reply:
Fixed (?) in 'Address review' patch. Ready for re-review.